pyproject.toml

```diff
@@ -1,4 +1,126 @@
-[tool.black]
+[build-system]
+requires = ["setuptools>=61.0", "wheel"]
+build-backend = "setuptools.build_meta"
+
+[project]
+name = "yasa"
+description = "YASA: Analysis of polysomnography recordings."
+readme = "README.rst"
+license = {text = "BSD (3-clause)"}
+authors = [
+    {name = "Raphael Vallat", email = "raphaelvallat9@gmail.com"},
+    {name = "Remington Mallett", email = "mallett.remy@gmail.com"},
+]
+maintainers = [
+    {name = "Raphael Vallat", email = "raphaelvallat9@gmail.com"},
+    {name = "Remington Mallett", email = "mallett.remy@gmail.com"},
+]
+classifiers = [
+    "Intended Audience :: Science/Research",
+    "Operating System :: MacOS",
+    "Operating System :: POSIX",
+    "Operating System :: Unix",
+    "Programming Language :: Python :: 3.9",
+    "Programming Language :: Python :: 3.10",
+    "Programming Language :: Python :: 3.11",
+    "Programming Language :: Python :: 3.12",
+]
+dynamic = ["version"]
+requires-python = ">=3.8"
+dependencies = [
+    "numpy>=1.18.1",
+    "scipy",
+    "pandas",
+    "matplotlib",
+    "seaborn",
+    "mne>=1.3",
+    "numba>=0.57.1",
+    "antropy",
+    "scikit-learn",
+    "tensorpac>=0.6.5",
+    "pyriemann>=0.2.7",
+    "sleepecg>=0.5.0",
+    "lspopt",
+    "ipywidgets",
+    "joblib",
+    "lightgbm",
+]
+
+[project.optional-dependencies]
+test = [
+    "pytest>=6",
+    "pytest-cov",
+    # Ensure coverage is new enough for `source_pkgs`.
+    "coverage[toml]>=5.3",
+    "ruff"
+]
+docs = [
+    "sphinx>7.0.0",
+    "pydata_sphinx_theme",
+    "numpydoc",
+    "sphinx-copybutton",
+    "sphinx-design",
+    "sphinx-notfound-page",
+]
+
+[project.urls]
+Homepage = "https://github.com/raphaelvallat/yasa/"
+Downloads = "https://github.com/raphaelvallat/yasa/"
+
+[tool.setuptools]
+py-modules = ["yasa"]
+include-package-data = true
+
+[tool.setuptools.package-data]
+yasa = [
+    "classifiers/*.joblib",
+]
+
+[tool.setuptools.packages.find]
+namespaces = false
+where = ["yasa"]
+
+[tool.setuptools.dynamic]
+version = {attr = "yasa.__version__"}
+
+[tool.pytest.ini_options]
+minversion = "6.0"
+addopts = "--showlocals --durations=10 --maxfail=2 --cov"
+doctest_optionflags= ["NORMALIZE_WHITESPACE", "IGNORE_EXCEPTION_DETAIL"]
+filterwarnings = [
+    "ignore::UserWarning",
+    "ignore::RuntimeWarning",
+    "ignore::FutureWarning",
+]
+markers = ["slow"]
+
+[tool.coverage.run]
+branch = true
+omit = [
+    "*/tests/*",
+]
+source_pkgs = ["yasa"]
+
+[tool.coverage.paths]
+source = ["yasa"]
+
+[tool.coverage.report]
+show_missing = true
+# sort = "Cover"
+
+[tool.ruff]
```

Comment:
Current changes work. But we could be more explicit with the exclusions and add accomodations for numpy-style docstrings. Just copy/paste some parts from this example:

```
[tool.ruff]
line-length = 100
select = ["E", "F", "I", "W", "C90"]

[tool.ruff.lint.per-file-ignores]
"__init__.py" = ["F401", "F403"]  # Ignore star and unused import violations for __init__.py files

[tool.ruff.lint.pydocstyle]
convention = "numpy"

[tool.ruff.format]
docstring-code-format = true
docstring-code-line-length = 20
```

Reply:
Nice thank you!